### PR TITLE
Refactor String Documentation

### PR DESCRIPTION
**Pull request by mcdobe100**
String will no longer appear as a "record" in documentation and changed many string methods to secondary methods so their documentation would still appear.

### CODE REVIEW

**Comment by e-kayrakli** [comment]
I'll give a more detailed review later today, but few quick notes:

- You see an example of CI failures on doc changes where it says "All checks have failed"
  - The reason for that is we shouldn't be referring to `String.string` as a record in `bytes` docs but should be referring to the module as a whole. An example of this is [here](https://github.com/chapel-lang/chapel/blob/master/modules/internal/Bytes.chpl#L40). ":mod:`String`" should be the way to go, going forward.

- Which also brings up the point that similar changes need to be done for the `bytes` implementation in `modules/internal/Bytes.chpl`. You can start doing that as part of this PR, wait until you get your first feedback, or leave it altogether for another follow up PR (which I'd hope would come soon after this one)

**Comment by bradcray** [comment]
> Which also brings up the point that similar changes need to be done for the bytes implementation in modules/internal/Bytes.chpl.

I don't think this PR should feel the need to convert Bytes similarly per se (in that it isn't something I asked Ben to do).

**Comment by bradcray** [comment]
I've just invited @mcdobe100 to be a member of the Chapel repo as the first step at clearing up the `cla required` label.

File: modules/internal/String.chpl [hunk 1]
```diff
@@ -53,15 +53,15 @@ within strings.
...
+This module supports casts from :mod:`String` to numeric types. Such casts
 will convert the string to the numeric type and throw an error if the string
 is invalid. For example:
 
 .. code-block:: chapel
 
   var number = "a":int;
 
-throws an error when it is executed, but
+  throws an error when it is executed, but
```
Comment (e-kayrakli) [inline modules/internal/String.chpl:64]: It's a good idea to preserve the whitespace

File: modules/internal/String.chpl [hunk 2]
```diff
@@ -832,23 +832,17 @@ module String {
   // in when possible.
   pragma "ignore noinit"
   pragma "no default functions" // avoid the default (read|write)This routines
-  record _string {
-    pragma "no doc"
+  pragma "no doc"
+  record _string { //Opening brace
```
Comment (e-kayrakli) [inline modules/internal/String.chpl:836]: Forgotten comment? Let's drop it

**Comment by e-kayrakli** [comment]
Test status:
- [ ] standard
- [ ] gasnet

**Comment by bradcray** [comment]
I ran a standard test on the branch last week, but before the final few commits were made (so it probably is worth re-running them, and I'd be happy to help with that until @mcdobe100 gets access to the test systems... I don't have much else testing in my workspaces currently).

**Comment by e-kayrakli** [comment]
I am running them already. Thanks!

**Comment by bradcray** [comment]
I was apparently misremembering and had to remove the "CLA required" manually, but happily, @mcdobe100's new PR didn't get similarly tagged, so we should be good from here on out.

**Comment by bradcray** [comment]
Woohoo, congrats @mcdobe100!  First commit in less than a week's work!

**Pull request merged**